plugins/rapid7_metasploit/komand_rapid7_metasploit/actions/search_for_exploit/action.py

```diff
@@ -14,43 +15,44 @@ def __init__(self):
         )
 
     def run(self, params={}):
-        search_term = params["search_term"].lower().encode("utf-8", "ignore")
+        search_term = params["search_term"].lower()
         results = {"exploits": [], "auxiliaries": [], "post_exploits": []}
         client = self.connection.client
         try:
             exploits = client.modules.exploits
             auxiliary = client.modules.auxiliary
             post_exploits = client.modules.post
 
-            for exploit in exploits:
-                current_exploit = client.modules.use("exploit", exploit)
-                description = current_exploit.description
-                if search_term.lower() in exploit or search_term.lower() in description.lower():
-                    results["exploits"].append(exploit)
-
-            for aux in auxiliary:
-                current_aux = client.modules.use("auxiliary", aux)
-                description = current_aux.description
-                try:
-                    if search_term in aux or search_term in description.lower():
-                        results["auxiliaries"].append(aux)
-                except UnicodeDecodeError as e:
-                    self.logger.error("Error: %s" % e)
-            for post in post_exploits:
-                current_post = client.modules.use("post", post)
-                description = current_post.description
-                if search_term.lower() in post or search_term.lower() in description.lower():
-                    results["post_exploits"].append(post)
-            return {"search_results": results}
+            self.do_section(client, search_term, results, exploits, "exploit", "exploits")
+            self.do_section(client, search_term, results, auxiliary, "auxiliary", "auxiliaries")
+            self.do_section(client, search_term, results, post_exploits, "post", "post_exploits")
+
         except Exception as e:
             self.logger.error("Error occurred while retrieving exploits: ", e)
             raise
 
+        return results
+
     def test(self):
         client = self.connection.client
         try:
             exploits = client.modules.exploits
+            # good for logging test results and also silences unused variable from prospector
+            self.logger.info(exploits)
         except Exception as e:
             self.logger.error("Error occurred while testing search for exploits: ", e)
             raise
         return {"success": True}
+
+    def do_section(self, client, search_term, results, list_in: list, list_name: str, dictionary_key: str):
```

Comment:
Can you typehint `client`, `search_term`, and `results`?

Reply:
Done!